ssas/service/server.go

```diff
@@ -257,6 +256,31 @@ func (s *Server) VerifyToken(tokenString string) (*jwt.Token, error) {
 	return jwt.ParseWithClaims(tokenString, &CommonClaims{}, keyFunc)
 }
 
+func (s *Server) VerifyClientSignedToken(tokenString string, trackingId string) (*jwt.Token, error) {
+	keyFunc := func(token *jwt.Token) (interface{}, error) {
+		if _, ok := token.Method.(*jwt.SigningMethodRSA); !ok {
+			return nil, fmt.Errorf("unexpected signing method: %v", token.Header["alg"])
+		}
+		claims := token.Claims.(*CommonClaims)
+		if claims.Issuer == ""{
+			return nil, fmt.Errorf("missing issuer (iss) in jwt claims")
+		}
+		system,err := ssas.GetSystemByID(claims.Issuer)
+		if err !=nil {
+			return nil, err
+		}
+		key, err := system.GetEncryptionKey(trackingId)
+		if err !=nil {
+			return nil, fmt.Errorf("key not found for system: %v", claims.Issuer)
+		}
+		pubKey,err := ssas.ReadPublicKey(key.Body)
+		if err !=nil {
+			return nil, err
+		}
+		return pubKey, nil
```

Comment:
Same as above (return a helpful generic message)

Reply:
not sure why this does not show up as outdated, but its been updated.